lib/charms/operator_libs_linux/v2/snap.py

```diff
@@ -273,7 +273,10 @@ def _snap(self, command: str, optargs: Optional[Iterable[str]] = None) -> str:
           SnapError if there is a problem encountered
         """
         optargs = optargs or []
-        _cmd = ["snap", command, self._name, *optargs]
+        if command == "alias":
+            _cmd = ["snap", command, *optargs]
+        else:
+            _cmd = ["snap", command, self._name, *optargs]
```

Comment:
Maybe I should create a separate method for this instead?

Reply:
Yeah, I think so. This is an awkward special case. The `_snap` method is intended to run a snap-specific operation on a named snap. I would probably just put the `subprocess.run` call in `alias` itself.

Reply:
There might also be a way to unify the various `subprocess.run` / `subprocess.check_output` calls so as to avoid repeating the `try/except` handling, but I don't know if it's worth it -- see what you think.

Reply:
> There might also be a way to unify the various `subprocess.run` / `subprocess.check_output` calls so as to avoid repeating the `try/except` handling, but I don't know if it's worth it -- see what you think.

I don't think It's worth it for just a single call with try catch.